src/frontend/screens/Settings/components/SettingsModal/index.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { AppSettings, GameInfo } from 'common/types'
+import { UpdateComponent } from 'frontend/components/UI'
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader
+} from 'frontend/components/UI/Dialog'
+import ContextProvider from 'frontend/state/ContextProvider'
+import { GamesSettings } from '../../sections'
+import { writeConfig } from 'frontend/helpers'
+import SettingsContext from '../../SettingsContext'
+import { SettingsContextType } from 'frontend/types'
+
+type Props = {
+  gameInfo: GameInfo
+}
+
+function SettingsModal({ gameInfo }: Props) {
+  const { setIsSettingsModalOpen, platform } = useContext(ContextProvider)
+
+  const [title, setTitle] = useState('')
+
+  const [currentConfig, setCurrentConfig] = useState<Partial<AppSettings>>({})
+
+  const { app_name: appName, runner } = gameInfo
+  const isLinux = platform === 'linux'
+  const isMac = platform === 'darwin'
+  const isMacNative = isMac && (gameInfo?.is_mac_native || false)
+  const isLinuxNative = isLinux && (gameInfo?.is_linux_native || false)
+
+  // Load Heroic's or game's config, only if not loaded already
+  useEffect(() => {
+    const getSettings = async () => {
+      const config = await window.api.requestGameSettings(appName)
+      setCurrentConfig(config)
+
+      setTitle(gameInfo?.title ?? appName)
+    }
+    getSettings()
+  }, [appName])
+
+  if (!title) {
+    return <UpdateComponent />
+  }
+
+  // create setting context functions
+  const contextValues: SettingsContextType = {
```

Comment:
most of this is duplicated with the Settings screen, since settings are kinda critical maybe we can move most of this into a `hasSettingsContext` hook to make sure we don't get this logic out of sync?

Reply:
I was wondering on how to avoid this duplication. My first idea was to use React Portals to render this settings but won't work since it needed the context to be passed on. 
How do you think this context would work? I wonder if we put the settings context above APP if the portal would work or at least the values would be available imediatelly.

Reply:
Done, created a hook that generate the context values and can be used anywhere.